core/devmode/src/main/java/io/quarkus/dev/ClassLoaderCompiler.java

```diff
@@ -104,6 +104,10 @@ public ClassLoaderCompiler(ClassLoader classLoader,
                                 for (String i : classPath.toString().split(" ")) {
                                     File f;
                                     try {
+                                        if (i.startsWith("file:")) {
+                                            i = i.substring(5, i.lastIndexOf('!'));
```

Comment:
Hello @dufoli , won't this cause a problem is the `!` character isn't part of the `String` `i`, in which case `lastIndexOf` will return -1?